tests/test_samples.py

```diff
@@ -648,6 +648,45 @@ def test_posterior_points():
     assert_array_equal(ns.posterior_points(0.5), ns.posterior_points(0.5))
 
 
+def test_importance_samples():
+    np.random.seed(3)
+    ns0 = NestedSamples(root='./tests/example_data/pc')
+
+    with pytest.raises(NotImplementedError):
+        ns0.importance_sample(ns0.logL, action='spam')
+
+    ns1 = ns0.importance_sample(ns0.logL, action='replace')
+    assert_array_equal(ns0.logL, ns1.logL)
+    assert_array_equal(ns0.logL_birth, ns1.logL_birth)
+    assert_array_equal(ns0.weights, ns1.weights)
+
+    ns1 = ns0.importance_sample(np.zeros_like(ns0.logL), action='add')
+    assert_array_equal(ns0.logL, ns1.logL)
+    assert_array_equal(ns0.logL_birth, ns1.logL_birth)
+    assert_array_equal(ns0.weights, ns1.weights)
+
+    mask = ((ns0.x0 > -0.1) & (ns0.x2 > 0.3) & (ns0.x4 < 3)).to_numpy()
+    ns1 = merge_nested_samples((ns0[mask], ))
+    logL_new = np.where(mask, ns0.logL, -np.inf)
+    ns2 = ns0.importance_sample(logL_new=logL_new)
+    assert_array_equal(ns1.logL, ns2.logL)
```

Comment:
If I've understood this test correctly, aren't ns1 and ns2 identical? If so, all of the tests below can be reduced to a single
`assert_array_equal(ns1, ns2)`. This would also make the checks against logZ and D trivially true.

It would be good to instead compare `NS1` to an `NS0` which then checks against the volume factors that you created in #120.

Reply:
> If I've understood this test correctly, aren't ns1 and ns2 identical?

I thought `importance_sample` with `-inf` would leave a different shape than direct masking, but no, they indeed are identical. Consequently, the structure of the rest of the test in relation to the volume factor tests, does not work the way I envisioned.

I fixed the tests accordingly.

Reply:
This was mentioned [here](https://github.com/williamjameshandley/anesthetic/pull/120#issuecomment-676537096) in #120, but I agree that this is unexpected. I'll raise a separate discussion issue now for how to resolve this.

Reply:
Yeah, I read that at first just as "produces the same result" and did not appreciate that it's actually identical.